src/ast2ram/utility/SipsMetric.cpp

```diff
@@ -310,8 +310,18 @@ std::vector<std::size_t> SelingerProfileSipsMetric::getReordering(
 /** Create a SIPS metric based on a given heuristic. */
 std::unique_ptr<SipsMetric> SipsMetric::create(const std::string& heuristic, const TranslationUnit& tu) {
     if (tu.global().config().has("auto-schedule")) {
-        return mk<SelingerProfileSipsMetric>(tu);
-    } else if (heuristic == "strict")
+        if (tu.getAnalysis<ast::analysis::ProfileUseAnalysis>().hasAutoSchedulerStats()) {
+            return mk<SelingerProfileSipsMetric>(tu);
+        } else {
+// TODO: enable this, but if we do the scheduler tests won't run as they do now
+#if 0
+            std::cerr << "WARNING: `--auto-schedule` cannot be used due to missing scheduler stats; falling back "
+                         "to heuristic '"
+                      << heuristic << "'" << ::std::endl;
+#endif
```

Comment:
should this TODO be addressed?

Reply:
I think we should print the warning, but I'm not too familiar with CMAKE to make the updates to the test scripts.

Reply:
Do you mind merging as is?